src/components/InfluenceChart/index.js

```diff
@@ -0,0 +1,751 @@
+// @flow
+
+/* eslint no-param-assign: off, no-param-reassign: off */
+
+import React from 'react'
+import moment from 'moment'
+import { connect } from 'react-redux'
+import { type SubjectId, type PersonAbstract, type PersonDetail } from '../../types'
+import { type PeopleCache } from '../../store'
+import { difference, union } from '../../utils/Set'
+
+const store = require('../../store')
+
+const d3 = require('d3')
+const fp = require('lodash/fp')
+
+const {
+  angleRadians,
+  largest,
+  populatePath,
+  radial,
+  smallest,
+} = require('../../util')
+
+const {
+  ALPHA,
+  ARROW_WIDTH,
+  BANNER_X,
+  BANNER_Y,
+  CHARGE_BASE,
+  CHARGE_HIDDEN,
+  CHARGE_RANDOM,
+  DEFAULT_ANIMATION_DURATION,
+  GRAVITY,
+  HEAD_ANGLE,
+  IMAGE_SIZE,
+  LINK_MIN_OFFSET,
+  LINK_RANDOM,
+  LINK_STRENGTH,
+  MARGIN,
+  NODE_SIZE,
+  TIMELINE_Y,
+} = require('../../constants')
+
+
+type LinkSegment = {
+  source: any,
+  target: any,
+}
+
+
+type Selection = {
+  append: any => Selection,
+  attr: (string, ?string | ?number | ?Function) => Selection,
+  call: (Function | Selection) => Selection,
+  classed: (string, bool | Function) => Selection,
+  data: (Array<any>, ?(any => any)) => Selection,
+  duration: number => Selection,
+  ease: number => Selection,
+  enter: () => Selection,
+  exit: () => Selection,
+  filter: Function => Selection,
+  on: (string, (any, number) => void) => Selection,
+  remove: () => Selection,
+  select: string => Selection,
+  selectAll: string => Selection,
+  style: (string, string) => Selection,
+  text: (string | Function) => Selection,
+  transition: (?number) => Selection,
+}
+
+type D3Scale<D, R> = {|
+  (D): D,
+  domain: Array<D> => void,
+  range: Array<R> => void,
+|}
+
+type ForceSimulation = {|
+  alpha: (?number) => number,
+  force: (string, any) => ForceSimulation,
+  nodes: Array<InvisibleNode | PersonNode> => ForceSimulation,
+  restart: () => void,
+  on: (string, () => void) => ForceSimulation,
+  restart: () => void,
+|}
+
+type LinkForces = {
+  links: Array<LinkSegment> => LinkForces,
+  strength: number => LinkForces,
+}
+
+
+type InvisibleNode = {
+  type: 'InvisibleNode',
+  x: number,
+  y: number,
+  vx: number,
+  vy: number,
+  getId: () => string,
+}
+
+type PersonNode = {
+  type: 'PersonNode',
+  x: number,
+  y: number,
+  vx: number,
+  vy: number,
+  person: PersonAbstract | PersonDetail,
+  getId: () => string,
+}
+
+
+type TLink = {
+  source: PersonNode,
+  middle: InvisibleNode,
+  target: PersonNode,
+}
+
+
+class TGraph {
```

Comment:
Can the graph types be parameterized so the graph can exist as a force graph without really knowing about its data types?